toolchain/parse/context.h

```diff
@@ -281,6 +281,13 @@ class Context {
   // Propagates an error up the state stack, to the parent state.
   auto ReturnErrorOnState() -> void { state_stack_.back().has_error = true; }
 
+  // Adds a node for a declaration's semicolon. Includes error recovery when the
```

Comment:
Maybe:
```suggestion
  // Adds a node with kind `node_kind` for a declaration's semicolon. Includes error recovery when the
```
It also would have been a bit less surprising to me if `node_kind` came earlier instead of after something only used as text when there is an error. You mention `AddNode` and `AddLeafNode` don't specifically document the `node_kind` parameter, but in those methods the `node_kind` parameter is first, right after the `Add...Node`.

Reply:
I've reordered parameters so that the diagnostic-specific things are last, does that work for you?

If not, I'll use your suggestion, but I'd like to be clear that saying "Adds a node kind `node_kind`" seems no less redundant.